### DISCUSSION TITLE
27

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#772] [Comments]
http://jojoldu.tistory.com/27

**Comment by jojoldu** [comment]
작성자: dkb

@Bean은 setter나 builder 등을 통해서 사용자가 프로퍼티를 변경해서 생성한 인스턴스를 스프링에게 관리하라고 맡기는것,
@Component는 클래스를 스프링더러 알아서 인스턴스 생성한 후에 등록(bean으로) 하라고 맡기는 것 차이가 있지요.

저도 구분없이 쓰다가 알게 되었더랍니다.

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

오!! dkb님 좋은 정보 감사합니다^^
몰랐던 내용인데 댓글주셔서 알게되었습니다 ㅎㅎ

**Comment by bell204** [comment]
1 @Bean- 개발자가 생성하지 않은 외부라이브러리 객체에 사용. 
2 @Component- 개발자가 직접 생성한 클래스에 사용.
이렇게 이해하면 되는것인가요?

**Comment by jeongmincha** [comment]
단순히 관례적으로만 Bean, Component 어노테이션의 사용을 구분하는 것으로 알고 있었는데,
실제 뜯어보니 Component는 class 위에 작성할 수 있고, Bean은 method 위에 작성할 수 있는 것이군요.
감사합니다 :)

**Comment by met-flo** [comment]
좋은 글 고맙습니다.
다만, 개발자가 직접 생성한 클래스라도 아래와 같이 직접 Bean 등록이 가능할 것 같은데요.
어떻게 생각하시는지요?

@Bean
public MyClass myClass() {
  return new MyClass();
}

**Comment by onjik** [comment]
엥 이게 맞나요?
개발자가 생성한 것도 @Configuration 안에 @Bean으로서 등록이 가능하고요.
@Component를 쓰냐 @Bean을 쓰냐의 기준을 단순히 개발자가 만들었냐 외부에서 만들었냐로 보는 것은 잘못된거 같습니다.
